tests/test_fixtures.py

```diff
@@ -32,19 +32,40 @@ def test_workspace_instance_scope_part2(workspace):
     assert workspace is workspace_instance[0]
 
 
-def test_hardware_node_add_remove(hardware):
+def _hardware_add_node(h, name, role) -> NodeBase:
     """
-    test the hardware fixture. Especially the handing of nodes
+    helper method to create a new hardware node
     """
-    nodes_length = len(hardware.nodes.keys())
+    nodes_length = len(h.nodes.keys())
     # create a new node
-    new_node = hardware.node_create('test1', NodeRole.WORKER, [])
+    new_node = h.node_create(name, role, [])
     # add the node to the hardware
-    hardware.node_add(new_node)
-    hardware.prepare_nodes(limit_to_nodes=[new_node])
+    h.node_add(new_node)
+    h.prepare_nodes(limit_to_nodes=[new_node])
     # we should have one more node now
-    assert len(hardware.nodes.keys()) == nodes_length+1
+    assert len(h.nodes.keys()) == nodes_length+1
+    return new_node
+
+
+def test_hardware_node_add_remove(hardware):
+    """
+    test the hardware fixture. Especially the handing of nodes
+    """
+    nodes_length = len(hardware.nodes.keys())
+    new_node = _hardware_add_node(hardware, 'test1', NodeRole.WORKER)
     # drop the node again
     hardware.node_remove(new_node)
     # we should have the old amount of nodes now
     assert len(hardware.nodes.keys()) == nodes_length
+
+
+def test_kubernetes_node_join(kubernetes):
+    # current number of nodes in the k8s cluster
+    kubernetes_nodes_length = len(kubernetes.v1.list_node().items)
+    # create a new node
+    new_node = _hardware_add_node(kubernetes.hardware, 'test1',
```

Comment:
This test leaves the cluster in a different state to which it receives it. We should probably note that so that subsequent tests know to expect 4 workers.

Similarly we could optimise the testing by having `test_hardware_node_add_remove` not remove the node again. It would, however, mean that we need to check if the node was added first in case somebody wanted to limit a test run to just this method.

Reply:
hm. not sure. I'm currently just running a single test from `test_fixtures.py` to verify that this part is working.
But in general I think a test should capture the current state, do modifications and check the expected state. And then maybe do a cleanup if needed.

If the test is blocking this PR, I'm happy to remove it again. I would like to get the other changes merged soon.

Reply:
If you could add a quick note about the dirty state the test leaves it in/a todo to clean up the cluster then I'm otherwise happy with the PR :-)

Reply:
done